File: src/utils/logger.rs
use console::style;

pub fn error(msg: &str) {
    println!("{} - {}", style("error").red(), msg);
}

pub fn info(msg: &str) {
    println!("{}  - {}", style("info").cyan(), msg)
}

pub fn ready(msg: &str) {
    println!("{} - {}", style("ready").green(), msg);
}

pub fn pick(msg: &str) {
    println!("{}  - {}", style("pick").color256(69), msg);
}

pub fn event(msg: &str) {
    println!("{} - {}", style("event").magenta(), msg);
}

pub fn full_info(msg: &str) {
    println!("{}  - {}", style("info").cyan(), style(msg).yellow());
}

pub fn warning(msg: &str) {
    println!("{}  - {}", style("warn").yellow(), msg);
}

File: scripts/logger.js
const colors = {
  red: '\x1b[31m',
  green: '\x1b[32m',
  yellow: '\x1b[33m',
  blue: '\x1b[34m',
  magenta: '\x1b[35m',
  cyan: '\x1b[36m',
  reset: '\x1b[0m',
};

class Style {
  constructor(text = '') {
    this.original = text;
    this.text = text;
  }

  apply(color) {
    this.text = color + this.text + colors.reset;
    return this;
  }

  red() {
    return this.apply(colors.red);
  }

  green() {
    return this.apply(colors.green);
  }

  yellow() {
    return this.apply(colors.yellow);
  }

  blue() {
    return this.apply(colors.blue);
  }

  magenta() {
    return this.apply(colors.magenta);
  }

  cyan() {
    return this.apply(colors.cyan);
  }

  color256(n) {
    this.text = `\x1b[38;5;${n}m` + this.text + colors.reset;
    return this;
  }

  bold() {
    this.text = '\x1b[1m' + this.text + colors.reset;
    return this;
  }

  toString() {
    return this.text;
  }
}

function style(text) {
  return new Style(text);
}

const logger = {
  info: (msg) => console.log(`${style('info').cyan()}  - ${msg}`),
  event: (msg) => console.log(`${style('event').color256(177)} - ${msg}`),
  error: (msg) => console.log(`${style('error').red()} - ${msg}`),
  ready: (msg) => console.log(`${style('ready').green()} - ${msg}`),
  warn: (msg) => console.log(`${style('warn').yellow()}  - ${msg}`),
};

module.exports = { logger, style };
